swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
 class Vendor:
-    pass
\ No newline at end of file
+    # Set up inventory
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    # Adds an item to the inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    # Removes an item from the vendors inventory
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        if item in self.inventory:
+            self.inventory.remove(item)
+        return item
+    
+    # Search through the vendor's inventory for an item with a matching id
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    #Check both inventories for items, swap items
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        # start swaping
+        self.remove(my_item)
+        other_vendor.add(my_item)
+
+        self.add(their_item)
+        other_vendor.remove(their_item)
```

Comment:
Not at all required, but rather a neat little trick: The remove method you created will return the item, which we could then append into our list all at once if we wanted to!

```suggestion
        other_vendor.add(self.remove(my_item))
        self.add(other_vendor.remove(their_item))
```

Reply:
I love this trick! I hadn’t thought about using the return value of my own remove() method like that—it’s really elegant!